chia/_tests/core/mempool/test_mempool_manager.py

```diff
@@ -2628,16 +2635,24 @@ async def test_create_block_generator(mempool_size: int, seed: int, transactions
     invariant_check_mempool(mempool_manager.mempool)
 
     assert mempool_manager.peak is not None
-    new_block_gen = await mempool_manager.create_block_generator(mempool_manager.peak.header_hash)
+    create_block = mempool_manager.create_block_generator if old else mempool_manager.create_block_generator2
+    new_block_gen = await create_block(mempool_manager.peak.header_hash, 10.0)
     assert new_block_gen is not None
 
     # now, make sure the generator we got is valid
 
-    assert len(expected_additions) == len(new_block_gen.additions)
-    assert expected_additions == set(new_block_gen.additions)
-    assert len(expected_removals) == len(new_block_gen.removals)
-    assert expected_removals == set(new_block_gen.removals)
-    assert expected_signature == new_block_gen.signature
+    if expect_failure:
+        assert len(expected_additions) != len(new_block_gen.additions)
+        assert expected_additions != set(new_block_gen.additions)
+        assert len(expected_removals) != len(new_block_gen.removals)
+        assert expected_removals != set(new_block_gen.removals)
+        assert expected_signature != new_block_gen.signature
```

Comment:
I imagine the best we can do when we expect failure, is checking for inequality?

Reply:
yeah, it's not so easy to predict exactly which transactions will make it